### PR TITLE
Add multi architecture image support

### LINKED ISSUE #282: Multi-architecture support
The idea is to build a multi-architecture image to support a common set of golang architectures (linux-only for now).

Question: For which kind of architectures do we see a benefit if we decide to implement that feature?

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
We now build a multi architecture container image in GCB for amd64 and arm64.
#### Which issue(s) this PR fixes:

Fixes #282 
#### Does this PR have test?
None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
Added multi-architecture support to the container image (amd64 and arm64 for now)
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
Did a manual test:
```

```
Logs: https://console.cloud.google.com/cloud-build/builds/967cc1f7-2ce1-4034-ad05-49af87bd49e4?project=448637284062

We now push 3 images:
- gcr.io/k8s-staging-sp-operator/security-profiles-operator (manifest list)
- gcr.io/k8s-staging-sp-operator/security-profiles-operator-amd64 (only amd64)
- gcr.io/k8s-staging-sp-operator/security-profiles-operator-arm64 (only arm64)

All three images have 3 tags: `latest`, a git ref and the `VERSION`:

**Comment by saschagrunert** [comment]
@hasheddan can you give this a look? :upside_down_face:

File: cloudbuild.yaml [hunk 1]
```diff
@@ -3,17 +3,22 @@
...
+      make image-cross
 substitutions:
   # _GIT_TAG will be filled with a git-based tag for the image, of the form
   # vYYYYMMDD-hash, and can be used as a substitution
   _GIT_TAG: '12345'
 images:
-  - 'gcr.io/$PROJECT_ID/security-profiles-operator:$_GIT_TAG'
-  - 'gcr.io/$PROJECT_ID/security-profiles-operator:latest'
+  - 'gcr.io/$PROJECT_ID/security-profiles-operator-amd64:$_GIT_TAG'
+  - 'gcr.io/$PROJECT_ID/security-profiles-operator-amd64:latest'
```
Comment (hasheddan) [inline cloudbuild.yaml:24]: @saschagrunert I didn't quite follow this change, could you provide some more context?

Comment (saschagrunert) [reply cloudbuild.yaml:24]: This is just a check after the cloudbuild.yaml has run. It verifies that those images exist in the registry, nothing more.

Comment (hasheddan) [reply cloudbuild.yaml:24]: Gotcha, thanks! That makes sense, wasn't aware it didn't work with manifests :)

**Review by hasheddan — APPROVED** [review]
/lgtm

**Review by jhrozek — APPROVED** [review]
Looks good (not adding lgtm as another review was requested)

File: cloudbuild.yaml (hunk 1 above)
Comment (jonjohnsonjr) [reply cloudbuild.yaml:24]: > It verifies that those images exist in the registry, nothing more.

Doesn't this actually _push_ those images from the docker daemon to the registry? https://cloud.google.com/build/docs/build-config#images

In this case, if you've already pushed them, I guess the push is a no-op?

Comment (saschagrunert) [reply cloudbuild.yaml:24]: That’s not doing the actual push. As I tested it, it was just checking that the images are now available (aka already pushed).

**Pull request merged**